torch/csrc/utils/python_dispatch.cpp

```diff
@@ -50,6 +54,70 @@ inline torch::CppFunction dispatch_str(const char* key, Func&& raw_f) {
   }
 }
 
+class PythonKernelHolder : public c10::OperatorKernel {
+  SafePyObject* func_;
+public:
+  PythonKernelHolder(py::object func) : func_(new SafePyObject(func.release().ptr(), getPyInterpreter())) {}
+  // This is a generally useful pattern and safer than directly using pybind11's
+  // py::object destructor.  This is because this object may outlive
+  // libtorch_python, so we want to disarm the deallocation if that happens.
+  // PyInterpreter does this correctly, pybind11 does not.
+  ~PythonKernelHolder() {
+    getPyInterpreter()->decref(func_->ptr(getPyInterpreter()), /*is_tensor*/ false);
+  }
+
+  void operator()(const c10::OperatorHandle& op, c10::DispatchKeySet keyset, torch::jit::Stack* stack) {
+    const auto& schema = op.schema();
+    const auto num_returns = schema.returns().size();
+
+    const auto num_arguments = schema.arguments().size();
+    auto arguments = torch::jit::pop(*stack, num_arguments);
+
+    // TODO: Some duplication with torch/csrc/autograd/python_variable.cpp
+
+    py::gil_scoped_acquire g;
+
+    // Pre-scan for arguments that match defaults
+    int64_t default_suffix_len = 0;
+    for (auto idx = arguments.size() - 1; idx >= 0; idx--) {
+      const auto& arg = schema.arguments()[idx];
+      if (!arg.default_value().has_value()) {
+        break;
+      }
+      const auto& default_ivalue = *arg.default_value();
+      const auto& ivalue = arguments[idx];
+      if (default_ivalue != ivalue) {
+        break;
+      }
+      default_suffix_len++;
+    }
+
+    auto args = py::reinterpret_steal<py::object>(PyTuple_New(num_arguments - default_suffix_len));
+
+    // TODO: actually populate kwargs sometimes?  At the moment, every argument
+    // just gets passed positionally
+    py::dict kwargs;
```

Comment:
So, it would be very good to deduplicate this code with torch/csrc/autograd/python_variable.cpp. In particular, we actually fixed the logic in python_variable.cpp to properly put kwarg only arguments in the kwargs dict. Since this is a copy paste you didn't get that fix. Putting this all in a helper function would solve that problem.